include/oneapi/tbb/partitioner.h

```diff
@@ -90,7 +92,8 @@ class affinity_partitioner_base: no_copy {
     /** Retains values if resulting size is the same. */
     void resize(unsigned factor) {
         // Check factor to avoid asking for number of workers while there might be no arena.
-        unsigned max_threads_in_arena = static_cast<unsigned>(max_concurrency());
+        unsigned max_threads_in_arena = unsigned(std::min(static_cast<std::size_t>(max_concurrency()),
+            tbb::global_control::active_value(tbb::global_control::max_allowed_parallelism)));
```

Comment:
Looks a little bit clumsy, don't you think? Does it make sense to move calculation of `maximum_arena_concurrency` into some dedicated function that could be reused in `get_initial_auto_partitioner_divisor` as well?

Reply:
I'm not sure. if need to introduce new method for these 2 lines. (we check not arena concurrency `min(arena_concurrency, allowed_concurrency)` but rather available amount of workers).

Reply:
I agree to move it into something like `get_num_possible_workers()` to highlight that the solution is based on the immediate value of workers as `std::min(...)` does not mean much to reader.

Reply:
Remove from affinity